lib/segment/src/vector_storage/query_scorer/metric_query_scorer.rs

```diff
@@ -30,8 +30,9 @@ impl<
         query: TypedDenseVector<VectorElementType>,
         vector_storage: &'a TVectorStorage,
     ) -> Self {
+        let preprocessed_vector = TMetric::preprocess(query);
         Self {
-            query: TMetric::preprocess(query),
+            query: TElement::slice_from_float_cow(&preprocessed_vector).to_vec(),
```

Comment:
It is good to know, that Cow won't actually invoke copy for owned vectors

Reply:
Fixed: https://github.com/qdrant/qdrant/pull/4049/commits/0edd6304ebbcecfe2bb52b428f464e70d04306f5
Now we don't do unnecessary copying. In this place for float and non-cosine distance there is no any reallocation anymore